src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/ClientErrorResultFilter.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Mvc.Infrastructure
+{
+    internal class ClientErrorResultFilter : IAlwaysRunResultFilter, IOrderedFilter
+    {
+        private readonly IDictionary<int, Func<ActionContext, IActionResult>> _clientErrorFactory;
+        private readonly ILogger<ClientErrorResultFilter> _logger;
+
+        /// <summary>
+        /// Gets the filter order. Defaults to -1000 so that it runs early.
+        /// </summary>
+        public int Order => -1000;
```

Comment:
Are there other result filters we have. Normally -1000 is a value we'd use by default for a filter we write. I think this needs to have a much earlier order.